EventHubs/tests/test_eventhub_metrics.py

```diff
@@ -1,48 +1,92 @@
+import sys
 import unittest
-import datetime
+from io import StringIO
+from datetime import timedelta, datetime
 from baseeventhubtest import BaseEventHubTest
-
-
+from azure.monitor.query import LogsQueryClient, LogsQueryStatus
 class TestEventHubMetrics(BaseEventHubTest):
 
     def setUp(self):
         super(TestEventHubMetrics, self).setUp()
-        self.RESOURCE_GROUP_NAME = "TestEventHubMetrics-%s" % (
-            datetime.datetime.now().strftime("%d-%m-%y-%H-%M-%S"))
-        self.function_name_prefix = "EventHubs_Metrics"
-        self.STORAGE_ACCOUNT_NAME = "sumometlogs"
-        self.template_name = 'azuredeploy_metrics.json'
-        self.log_table_name = "AzureWebJobsHostLogs%d%02d" % (
-            datetime.datetime.now().year, datetime.datetime.now().month)
-        self.event_hub_namespace_prefix = "SumoMetricsNamespace"
-        self.eventhub_name = 'insights-metrics-pt1m'
-
+        self.RESOURCE_GROUP_NAME = "EventHubMetrics-%s" % (str(int(datetime.now().timestamp())))
+        self.template_name = "azuredeploy_metrics.json"
+        self.event_hub_namespace_prefix = "SMNamespace"
+        self.eventhub_name = "insights-metrics-pt1m"
+        
     def test_pipeline(self):
-
         self.create_resource_group()
         self.deploy_template()
-        print("Testing Stack Creation")
-        self.assertTrue(self.resource_group_exists(self.RESOURCE_GROUP_NAME))
-        self.table_service = self.get_table_service()
-        self.insert_mock_logs_in_EventHub('metrics_fixtures.json')
-        self.check_error_logs()
+        self.assertTrue(self.resource_group_exists(self.RESOURCE_GROUP_NAME)) 
+        self.insert_mock_metrics_in_EventHub('metrics_fixtures.json')
+        self.check_success_log()
+        self.check_error_log()
+        self.check_warning_log()
+    
+    def fetchlogs(self, query):
+        # Save the original stdout for later comparison
+        original_stdout = sys.stdout
+        sys.stdout = StringIO()
+
+        try:
+            client = LogsQueryClient(self.azure_credential)
+            response = client.query_workspace('800e2f15-1fa6-4c5a-a3f5-db8f647ee8a1', query, timespan=timedelta(days=1))
+                
+            if response.status == LogsQueryStatus.PARTIAL:
+                error = response.partial_error
+                data = response.partial_data
+            elif response.status == LogsQueryStatus.SUCCESS:
+                data = response.tables
+            for table in data:
+                for col in table.columns:
+                    print(col + "    ", end="")
+                for row in table.rows:
+                    for item in row:
+                        print(item, end="")
+                    print("\n")
+        except Exception as e:
+            print("An unexpected error occurred during the test:")
+            print("Exception", e)
+
+        output = sys.stdout.getvalue()
```

Comment:
as discussed move this in fetch logs function

Reply:
Currently, it is the part of "fetch logs" function